src/libdredd/src/mutation_replace_unary_operator.cc

```diff
@@ -240,9 +273,18 @@ void MutationReplaceUnaryOperator::GenerateUnaryOperatorReplacement(
     }
     mutant_offset++;
   }
-  new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
-               << mutant_offset << ")) return " + arg_evaluated + ";\n";
-  mutant_offset++;
+
+  if (!optimise_mutations ||
+      MutationReplaceExpr::ExprIsEquivalentTo(*unary_operator_.getSubExpr(), 0,
+                                              ast_context) ||
+      MutationReplaceExpr::ExprIsEquivalentTo(*unary_operator_.getSubExpr(), 1,
+                                              ast_context) ||
+      MutationReplaceExpr::ExprIsEquivalentTo(*unary_operator_.getSubExpr(), -1,
+                                              ast_context)) {
+    new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                 << mutant_offset << ")) return " + arg_evaluated + ";\n";
+    mutant_offset++;
+  }
 }
```

Comment:
A comment here to justify what these are covered by would be good.

Reply:
Done.